From d6d38572b7df5c3298e38f6f0818ffab0123b4c7 Mon Sep 17 00:00:00 2001
From: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Date: Tue, 30 Aug 2022 15:55:06 +0300
Subject: [PATCH 36/42] ddr: Move s32cc platform includes to ddr_plat.h

Moved the platform includes to platform-specific ddr_plat.h,
which is to be included also in the Custom DDR Drivers that
will be supported with a future commit.

Issue: ALB-9210
Upstream-Status: Pending 

Signed-off-by: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 include/drivers/nxp/s32/ddr/ddr_utils.h     | 13 +------------
 include/drivers/nxp/s32/ddr/s32g/ddr_plat.h | 15 +++++++++++++++
 include/drivers/nxp/s32/ddr/s32r/ddr_plat.h | 14 ++++++++++++++
 plat/nxp/s32/s32g/s32g_common.mk            |  1 +
 plat/nxp/s32/s32r/s32r.mk                   |  3 ++-
 5 files changed, 33 insertions(+), 13 deletions(-)
 create mode 100644 include/drivers/nxp/s32/ddr/s32g/ddr_plat.h
 create mode 100644 include/drivers/nxp/s32/ddr/s32r/ddr_plat.h

diff --git a/include/drivers/nxp/s32/ddr/ddr_utils.h b/include/drivers/nxp/s32/ddr/ddr_utils.h
index 6b3d3c275..52f994864 100644
--- a/include/drivers/nxp/s32/ddr/ddr_utils.h
+++ b/include/drivers/nxp/s32/ddr/ddr_utils.h
@@ -31,18 +31,7 @@
 #ifndef DDR_UTILS_H_
 #define DDR_UTILS_H_
 
-#include <stdbool.h>
-#if !defined(PLAT_s32r)
-#include <plat/nxp/s32g/bl31_ssram/ssram_mailbox.h>
-#else
-#include <arch_helpers.h>
-#include <stdint.h>
-#endif
-
-#if !defined(PLAT_s32r)
-/* Uncomment to store the CSR registers after executing DDR training */
-#define STORE_CSR_ENABLE
-#endif
+#include <ddr_plat.h>
 
 /* Possible errors */
 #define NO_ERR              0x00000000U
diff --git a/include/drivers/nxp/s32/ddr/s32g/ddr_plat.h b/include/drivers/nxp/s32/ddr/s32g/ddr_plat.h
new file mode 100644
index 000000000..0ba1f53cb
--- /dev/null
+++ b/include/drivers/nxp/s32/ddr/s32g/ddr_plat.h
@@ -0,0 +1,15 @@
+/*
+ * Copyright 2022 NXP
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef DDR_PLAT_H_
+#define DDR_PLAT_H_
+
+#include <stdbool.h>
+#include <plat/nxp/s32g/bl31_ssram/ssram_mailbox.h>
+
+#define STORE_CSR_ENABLE
+
+#endif /* DDR_PLAT_H_ */
diff --git a/include/drivers/nxp/s32/ddr/s32r/ddr_plat.h b/include/drivers/nxp/s32/ddr/s32r/ddr_plat.h
new file mode 100644
index 000000000..6400311f0
--- /dev/null
+++ b/include/drivers/nxp/s32/ddr/s32r/ddr_plat.h
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2022 NXP
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef DDR_PLAT_H_
+#define DDR_PLAT_H_
+
+#include <stdbool.h>
+#include <arch_helpers.h>
+#include <stdint.h>
+
+#endif /* DDR_PLAT_H_ */
diff --git a/plat/nxp/s32/s32g/s32g_common.mk b/plat/nxp/s32/s32g/s32g_common.mk
index ec52cdfb1..92fd5d791 100644
--- a/plat/nxp/s32/s32g/s32g_common.mk
+++ b/plat/nxp/s32/s32g/s32g_common.mk
@@ -22,6 +22,7 @@ include ${S32_SOC_FAMILY}/bl31_ssram/bl31_ssram.mk
 PLAT_INCLUDES		+= -I${S32_SOC_FAMILY}/include \
 			   -I${S32_SOC_FAMILY}/bl31_sram/include \
 			   -I${S32_SOC_FAMILY}/bl31_ssram/include \
+			   -Iinclude/${S32_DRIVERS}/ddr/s32g \
 
 PLAT_BL_COMMON_SOURCES	+= \
 			   ${S32_SOC_FAMILY}/s32g_mc_me.c \
diff --git a/plat/nxp/s32/s32r/s32r.mk b/plat/nxp/s32/s32r/s32r.mk
index 8ecbaed74..b0bf18540 100644
--- a/plat/nxp/s32/s32r/s32r.mk
+++ b/plat/nxp/s32/s32r/s32r.mk
@@ -17,7 +17,8 @@ include plat/nxp/s32/s32_common.mk
 PLAT_SOC_PATH	:= ${S32_PLAT}/${S32_PLAT_SOC}
 
 PLAT_INCLUDES	+=	-I${PLAT_SOC_PATH}/include \
-					-I${S32_PLAT}/include \
+			-I${S32_PLAT}/include \
+			-Iinclude/${S32_DRIVERS}/ddr/s32r \
 
 PLAT_BL_COMMON_SOURCES += ${S32_DRIVERS}/clk/s32r45_clk.c \
 		${S32_PLAT}/s32gen1_mc_me.c \
-- 
2.17.1

